101-vm-userdata/README.md

```diff
@@ -0,0 +1,16 @@
+# Deploy a Virtual Machine with UserData
+
+![Azure Public Test Date](https://azurequickstartsservice.blob.core.windows.net/badges/101-vm-userdata/PublicLastTestDate.svg)
+![Azure Public Test Result](https://azurequickstartsservice.blob.core.windows.net/badges/101-vm-userdata/PublicDeployment.svg)
+
+![Azure US Gov Last Test Date](https://azurequickstartsservice.blob.core.windows.net/badges/101-vm-userdata/FairfaxLastTestDate.svg)
+![Azure US Gov Last Test Result](https://azurequickstartsservice.blob.core.windows.net/badges/101-vm-userdata/FairfaxDeployment.svg)
+
+![Best Practice Check](https://azurequickstartsservice.blob.core.windows.net/badges/101-vm-userdata/BestPracticeResult.svg)
+![Cred Scan Check](https://azurequickstartsservice.blob.core.windows.net/badges/101-vm-userdata/CredScanResult.svg)
+
+[![Deploy To Azure](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/deploytoazure.svg?sanitize=true)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F101-vm-userdata%2Fazuredeploy.json)  [![Visualize](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/visualizebutton.svg?sanitize=true)](http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F101-vm-userdata%2Fazuredeploy.json)
+
+This template allows you to create a Virtual Machine with User Data. This template also deploys a Storage Account, Virtual Network, Public IP addresses and a Network Interface.
+
+You can verify the Custom Data in the file `/var/lib/cloud/instance/user-data.txt`.
```

Comment:
this template is not related to customData, only userData, right?
so this line should be dropped?

Reply:
Missed that one. There's no equivalent file for user data correct?

Reply:
For right now (April 2021), at least, UserData will not be copied to a local file, so this template should not mention any local file.
For this GA, UserData is not being passed to CloudInit. This GA only enables UserData showing up in IMDS.

However, In July 2021 (or later), CloudInit is supposed to start using UserData.
Maybe CloudInit will copy UserData  to `/var/lib/cloud/instance/user-data.txt` in the future?

I am reading https://cloudinit.readthedocs.io/en/latest/topics/faq.html, but not clear on the details/expectations of CloudInit.